tests/io_tests/test_hdf5.py

```diff
@@ -7,6 +7,7 @@
 
 
 class TestHdf5:
+    @pytest.mark.skipif(os.name == 'nt', reason="Numpy default integer types differ on Windows.")
```

Comment:
How many test breaks? 
For `list`, `tuple`, `array` and `large list` you can change the integers to float.
For `heterogenous list` you can change integers to complex, str or None.

If then only `int` and `np int` fail, you could add if statements.